src/bindings/python/src/pyopenvino/core/core.cpp

```diff
@@ -148,12 +150,85 @@ void regclass_Core(py::module m) {
             :type model: openvino.runtime.Model
             :param device_name: Name of the device which will load the model.
             :type device_name: str
-            :param properties: Optional dict of pairs: (property name, property value) relevant only for this load operation.
+            :param properties: Optional dict of pairs: (property name, property value) relevant only for this load
+             operation.
             :type properties: dict
             :return: A compiled model.
             :rtype: openvino.runtime.CompiledModel
         )");
 
+    cls.def(
+        "compile_model",
+        [](ov::Core& self,
+           const std::shared_ptr<const ov::Model>& model,
+           const ov::RemoteContext& context,
+           const std::map<std::string, py::object>& properties) {
+            auto _properties = Common::utils::properties_to_any_map(properties);
+            py::gil_scoped_release release;
+            return self.compile_model(model, context, _properties);
+        },
+        py::arg("model"),
+        py::arg("context"),
+        py::arg("properties"),
+        R"(
+            Creates a compiled model from a source model object and loads it to device associated with context.
+            Users can create as many compiled models as they need, and use them simultaneously
+            (up to the limitation of the hardware resources).
+
+            GIL is released while running this function.
+
+            :param model: Model acquired from read_model function.
+            :type model: openvino.runtime.Model
+            :param context: Context which created on target device.
+            :type context: openvino.runtime.RemoteContext
```

Comment:
Regarding "init levels" -- this is example: there will be no `openvino.runtime.RemoteContext` available, only `openvino._pyopenvino.RemoteContext` will be exposed.